### PR TITLE
Update kafka example

### LINKED ISSUE #2587: Knative KafaSource detects wrong URL to serve events
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
- Deployed knative-eventing and kafkasource.
- Added https://github.com/kserve/kserve/blob/master/docs/samples/kafka/addressable-resolver.yaml
- knative eventing is able to read the kafka source.
- The service name created by inferenceservice is \<isvc-name>-predictor-default. However the kafkasource sends the events to http://\<isvc-name>.\<namespace>.svc.cluster.local

**What did you expect to happen:**
- I expected the requests to be sent to http://\<isvc-name>-predictor-default.\<namespace>.svc.cluster.local

**What's the InferenceService yaml:**
```
apiVersion: "serving.kserve.io/v1beta1"
kind: "InferenceService"
metadata:
  name: "<inference-name>"
  namespace: "\<namespace>"
  annotations:
    prometheus.io/scrape: 'true'
    prometheus.io/port: '8082'
  labels:
    name: "\<label>"
spec:
  predictor:
    minReplicas: 3
    maxReplicas: 100
    pytorch:
      name: \<name>
      storageUri: gs://<storage>
      resources:
        limits:
          cpu: 3000m
          memory: 3Gi
        requests:
          cpu: 2000m
          memory: 3Gi
```
**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
```
ingress: |-
    {
        "ingressGateway" : "knative-serving/knative-ingress-gateway",
        "ingressService" : "istio-ingressgateway.istio-system.svc.cluster.local",
        "localGateway" : "knative-serving/knative-local-gateway",
        "localGatewayService" : "knative-local-gateway.istio-system.svc.cluster.local",
        "ingressDomain"  : "example.com",
        "ingressClassName" : "kong",
        "domainTemplate": "{{ .Name }}-{{ .Namespace }}.{{ .IngressDomain }}",
        "urlScheme": "http"
    }
```

**Environment:**
 Using Kong
 RawDeployment
Torchserve version : 0.6.1
- Istio Version: Istio not installed
- Knative Version: knative serving not installled. knative eventing v1.8.2
- KServe Version: 0.9.0
- Kubeflow version: N/A
- Cloud Environment: GKE
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`): 1.23
- OS (e.g. from `/etc/os-release`): GKE

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:
Updated the kafka example

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2587 

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please update the mnist tranformer image in kserve docker repo.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

File: docs/samples/kafka/image_transformer/image_transformer.py [hunk 1]
```diff
@@ -41,11 +44,15 @@ def __init__(self, name: str, predictor_host: str):
...
 
-    def preprocess(self, inputs: Union[Dict, CloudEvent, InferRequest],
-                   headers: Dict[str, str] = None) -> Union[Dict, InferRequest]:
-        if inputs['EventName'] == 's3:ObjectCreated:Put':
-            bucket = inputs['Records'][0]['s3']['bucket']['name']
-            key = inputs['Records'][0]['s3']['object']['key']
+    async def preprocess(self, inputs: Union[Dict, CloudEvent, InferRequest],
+                         headers: Dict[str, str] = None) -> Union[Dict, InferRequest]:
+        logging.info("Received input %s", inputs)
+        event_data = await super().preprocess(inputs, headers) if inspect.iscoroutinefunction(self.preprocess) \
```
Comment (yuzisun) [inline docs/samples/kafka/image_transformer/image_transformer.py]: Depends on https://github.com/kserve/kserve/pull/2881 to remove the necessarity of calling super().preprocess.

Comment (sivanantha321) [reply docs/samples/kafka/image_transformer/image_transformer.py]: removed super().preprocess call

**Comment by sivanantha321** [comment]
@yuzisun We need to update the kserve/mnist-transformer image, once the PR #2881 is merged

**Pull request merged**